### PR TITLE
[DPR2-1521] refactor authCheck to only execute access level policies

### PR DESCRIPTION
**Pull request by djcostin**
- Updated remaining endpoints with authorisation policy check
- switched authCheck to only check access level policy types, was failing row level policies
- fixed tests adding default access level policy when needed

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportinglib/service/SyncDataApiService.kt [hunk 1]
```diff
@@ -109,19 +114,30 @@ class SyncDataApiService(
...
     )
+    checkAuth(productDefinition, userToken)
     val policyEngine = PolicyEngine(productDefinition.policy, userToken)
     return Count(
       configuredApiRepository.count(
         filters = validateAndMapFilters(productDefinition, filters, null),
         query = productDefinition.reportDataset.query,
         reportId = reportId,
-        policyEngineResult = policyEngine.execute(),
+        policyEngineResult = policyEngine.execute(PolicyType.ROW_LEVEL),
```
Comment (stalwartstu) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportinglib/service/SyncDataApiService.kt:124]: Do you also need Access here, too?

Comment (djcostin) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportinglib/service/SyncDataApiService.kt:124]: ah good spot, actually should just be execute(), looking at it now

Comment (djcostin) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportinglib/service/SyncDataApiService.kt]: updated that

**Review by stalwartstu — APPROVED** [review]

**Comment by gavriil-g-moj** [comment]
The single report definition endpoint doesn't seem to have the auth check as part of these changes. (edited)

**Pull request merged**